Allow patching of custom kernel tarballs

Copyright (C) 2017 Curt Brune <curt@cumulusnetworks.com>
Copyright (C) 2014 david_yang <david_yang@accton.com>

SPDX-License-Identifier:     GPL-2.0

diff --git a/scripts/build/kernel/linux.sh b/scripts/build/kernel/linux.sh
index cd4a456..73dea7c 100644
--- a/scripts/build/kernel/linux.sh
+++ b/scripts/build/kernel/linux.sh
@@ -66,9 +66,9 @@ do_kernel_extract() {
     # tarball; in either case, we need to extract
     CT_Extract "linux-${CT_KERNEL_VERSION}"
 
-    # If using a custom tarball, no need to patch
+    # For ONIE, allow patching custom kernel tarballs
     if [ "${CT_KERNEL_LINUX_CUSTOM}" = "y" ]; then
-        return 0
+        rm -f ${CT_SRC_DIR}/.linux-${CT_KERNEL_VERSION}.patched
     fi
     CT_Patch "linux" "${CT_KERNEL_VERSION}"
 
